var/spack/repos/builtin/packages/py-pyshacl/package.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack.package import *
+
+
+class PyPyshacl(PythonPackage):
+    """A Python validator for SHACL."""
+
+    homepage = "https://github.com/RDFLib/pySHACL"
+    pypi = "pyshacl/pyshacl-0.17.2.tar.gz"
+
+    version("0.20.0", sha256="47f014c52cc69167b902c89b3940dd400f7f5d2169a62f97f837f3419b4a737d")
+    version("0.17.2", sha256="46f31c7a7f7298aa5b483d92dbc850ff79a144d26f1f41e83267ed84b4d6ae23")
+
+    depends_on("py-poetry", type="build")
```

Comment:
```suggestion
    depends_on("python@3.7:3", type=("build", "run"))
    depends_on("py-poetry-core@1.1:1", type="build")
```

Reply:
I notice the python dependency is often not specified (I guess it's implied by inheriting from PythonPackage). Do you add it here explicitly because of [the `Python :: 3 :: Only` specification](https://github.com/RDFLib/pySHACL/blob/v0.20.0/pyproject.toml#L26)?

Reply:
Spack officially only supports Python 3.7+. So I usually don't add the Python version for something like `@2.7,3.5:` or `@3.5:` because it's implied. However, some day, Python 4 will come out, and I don't want to have to retroactively update all packages. Even if this package happened to work fine in Python 4, the build system says it only supports Python 3, so the build would still fail. So in summary, I only specify required Python versions when it's a subset of 3.7+ (e.g., `@3.8:`, `@:3`).

Reply:
Makes sense, thanks for clearing that up!